test/MainLangTests.hs

```diff
@@ -0,0 +1,51 @@
+--Tests for the type checker and evaluator in Core/MainLang.hs.
+module MainLangTests where
+
+import           Juvix.Core.MainLang
+
+import           Control.Monad.Except
+import           Numeric.Natural
+import           Prelude
+
+import           Test.Tasty
+import           Test.Tasty.HUnit
+import           Test.Tasty.QuickCheck
+
+import           Test.QuickCheck
+import           Test.QuickCheck.Gen
+
+--property tests of type checker:
+--the term's inferred type equals to the input type
+--property test of evaluator:
+-- \x.x (any term) evaluates to (any term evaluated)
+{-lamProp :: CTerm -> Env -> Property
+lamProp cterm env = App (cEval (Lam Bound 0) env) cterm == cterm-}
+-- any constant term evaluates to itself
+constProp ∷ CTerm → Env → Bool
+constProp (Star i) env = cEval (Star i) env == VStar i
+constProp Nats env     = cEval Nats env == VNats
+  --constProp _ _          = True --Not testing non-const terms
+
+{-TODO need to combine generators to generate CTerms http://hackage.haskell.org/package/QuickCheck-2.13.2/docs/Test-QuickCheck-Gen.html
```

Comment:
Some manual testcases for typechecking would be nice as well, e.g. the SKI combinators as in the Agda code.

Reply:
Will do in another PR.